### ISSUE TITLE
[BUG][Functional] Creating second VPC with same VPC prefix in the name fails

### ISSUE THREAD

**Issue by charliejllewellyn** [aws-samples/aws-secure-environment-accelerator#731]
Bug reports which fail to provide the required information will be closed without action.

**Required Basic Info**
- Accelerator Version:  v1.3.3
- Install Type: Clean
- Upgrade from version: N/A

**Describe the bug**
When specifying a second VPC in an account with the same Prefix e.g.

```
"vpc": [{
      "deploy": "local",
      "name": "External",
```

and

```
    {
      "deploy": "local",
      "name": "External-Firewall-Egress",
```

The accelerator fails at "Phase 2".

**Failure Info**
- What error messages have you identified, if any: 
The "Phase 2" CloudFormation stack reports:

```
Received response status [FAILED] from custom resource. Message returned: Error: Cannot put log group retention: {"message":"The specified log group does not exist.","code":"ResourceNotFoundException","time":"2021-05-12T06:52:01.476Z","requestId":"f65fb1ec-684f-4b7b-9cca-7eede3c2bae3","statusCode":400,"retryable":false,"retryDelay":74.83244952669233} (RequestId: bb27c5ae-54e2-4b95-93c2-895e9ccf7734)
```

- What symptoms have you identified, if any:
State machine fails

**Required files**
- Please provide a copy of your config.json file (sanitize if required)

**Steps To Reproduce**
1. Create account definition with two VPC's both starting with the same VPC prefix e.g. "External" and "External-something"
2. Deploy the accelerator
3. Phase 2 fails

**Expected behavior**
I expect two VPC's to be created

**Screenshots**

**Additional context**
The issue occurs because the CloudWatch log group fails to create if the "External-something" gets created first. This is because the code checks if the log group exists first base on the prefix. So if the "External-something" log group exists then the code describes log groups and filters by prefix. In the case of the second VPC this would be "External" so the log group matches and causes the install to believe it exists already. This causes the set log retention call to fail. See [this code](https://github.com/aws-samples/aws-secure-environment-accelerator/blob/c3ed6e05a0f2c159455d7a303245ee11120abc15/src/lib/custom-resources/cdk-logs-log-group/runtime/src/index.ts#L42).

**Workaround**
Start the VPC's with unique prefixes

**Comment by Brian969** [comment]
This has already been fixed in the pending v1.5.x release.  Can you work around the issue in the meantime?

**Comment by charliejllewellyn** [comment]
> This has already been fixed in the pending v1.5.x release. Can you work around the issue in the meantime?

Yep, can workaround :)

**Comment by Brian969** [comment]
Shipping in v1.5.0 (version # could change)
